client/src/components/hooks/useSquidDetails.ts

```diff
@@ -0,0 +1,20 @@
+import { useQuery, UseQueryResult } from "react-query";
+import { AxiosError } from "axios";
+
+import { ApiClient } from "../../backend/ApiClient";
+
+// DRY and re-use for multiple interfaces
+interface Squid {
+  name: string;
+  species: string;
+  experiencePoints?: number;
+  specialPower?: string;
+  imageUrl?: string;
+}
+
+export const useSquidDetails = (id): UseQueryResult<Squid, AxiosError> =>
+  useQuery(["squid", { id }], () =>
```

Comment:
Preferred convention: `["squids", id]`. This pattern makes it possible to be clever with how you invalidate queries, if you do so -- you could invalidate "squids" and invalidate both your squids index queries and your squid show page queries, or you could invalidate `["squids", id]` to invalidate only your show page query and any even more specific queries (`["squids", id, "weapons"]`) (query key matching when invalidating is loose by default). 

I personally prefer just using the id instead of `{ id }` here because with this pattern (which kinda emulates route nesting... /squids/3) it's obvious that the number is an id (in contrast to { page: 2, limit: 4, offset: 2 }, where the label is necessary to explain what the numeral is).